### DISCUSSION TITLE
No MySQL data after upgrade to Netdata 1.45

### DISCUSSION THREAD

**Discussion by aloz77** [netdata/netdata#17230] [Q&A] [answered]
On Debian 11 with MariaDB 10.11.7 after upgrade to Netdata 1.45 all MySQL data are not collected anymore.

![image](https://github.com/netdata/netdata/assets/9195877/35feffcc-83e5-4090-a88b-290a5861776d)

What can be the reason? How to debug?

**Comment by xPaw** [comment]
Seems like I have the same problem. If I restart netdata, it gets one data point and then no more data comes after.

**Reply by ilyam8** [reply]
Hey, @aloz77 @xPaw. What is your OS? Linux? For running apps auto-detection we switched from portscan to:

- localhost: reading /proc/net/{tcp,upd,tcp6,udp6}. Netdata creates a MySQL data collection job if the [app listens on 3306 or /proc/pid/comm is mysqld/mariadbd](https://github.com/netdata/netdata/blob/bf64532d402bae038aef4aa5e2d3ff4f29855403/src/go/collectors/go.d.plugin/config/go.d/sd/net_listeners.conf#L67-L68).
- docker: querying docker engine. Netdata creates a MySQL data collection job if the [container exposes port 3306 or the image matches the expected pattern](https://github.com/netdata/netdata/blob/f70556d7cb121178de9c01cf11b97c1956ad1221/src/go/collectors/go.d.plugin/config/go.d/sd/docker.conf#L45-L46).


Is your MySQL running on the host or is it a docker container?


> What can be the reason? How to debug?

Using [debug mode](https://github.com/netdata/netdata/tree/v1.45.0/src/go/collectors/go.d.plugin/modules/mysql#debug-mode). Share the output.

**Reply by aloz77** [reply]
MariaDB is running on the host. The host is Debian 11. Here is the debug log

[netdata-mysql-debug.log](https://github.com/netdata/netdata/files/14755171/netdata-mysql-debug.log)

**Reply by ilyam8** [reply]
I see

```
time=2024-03-26T08:31:45.795Z level=error msg="error on pinging the mysql database [netdata@tcp(127.0.0.1:3306)/]: Error 1045 (28000): Access denied for user 'netdata'@'127.0.0.1' (using password: NO)" plugin=go.d collector=mysql job=local
time=2024-03-26T08:31:45.795Z level=error msg="check failed" plugin=go.d collector=mysql job=local
```

Access denied for `netdata` user. Does it work if you set user to `root` and without a password? What credentials do you expect to be working?

**Reply by xPaw** [reply]
His log shows that it connects and gathers stats though, and then it stops.

I tried to run in debug mode, and I get the same thing, (I have a netdata@localhost user with no password as expected). It stops after this:
> INF module/job.go:267 stopped collector=mysql job=local
DBG mysql/mysql.go:124 using DSN [netdata@tcp(127.0.0.1:3306)/] collector=mysql job=local
ERR mysql/mysql.go:132 error on pinging the mysql database [netdata@tcp(127.0.0.1:3306)/]: Error 1130: Host '127.0.0.1' is not allowed to connect to this MariaDB server collector=mysql job=local
ERR module/job.go:238 check failed collector=mysql job=local

This doesn't seem correct as it collects stats before this happens, and it worked before updating netdata.

**Reply by ilyam8** [reply, marked as answer]
> His log shows that it connects and gathers stats though, and then it stops.

You are right! I didn't notice that. I know what the issue is. To fix it:

```bash
cd /etc/netdata
sudo ./edit-config go.d/mysql.conf
# just quit the editor, the script will copy "mysql.conf" to the user directory
sudo systemctl restart netdata
```

**Reply by aloz77** [reply]
Thank you @ilyam8, this fixed the issue.

**Reply by xPaw** [reply]
Will this be fixed without requiring a config change?
